File: src/main/java/org/minutebots/robot/Robot.java
package org.minutebots.robot;

import edu.wpi.first.wpilibj.TimedRobot;
import edu.wpi.first.wpilibj.command.Scheduler;
import org.minutebots.robot.hardwareconfigurations.*;
import org.minutebots.robot.subsystems.Drivetrain;
import org.minutebots.robot.utilities.VisionCommunication;

public class Robot extends TimedRobot {

  public static HardwareManger hardwareManager = new Asimov();
  public static boolean isAuto = false;
  

  @Override
  public void robotInit() {
    hardwareManager.init();
    hardwareManager.closeSolenoids();
  }

  @Override
  public void robotPeriodic() {
    OI.loop();
  }

  @Override
  public void disabledPeriodic(){
  }

  @Override
  public void autonomousInit() {
    isAuto = true;
    Drivetrain.getInstance().resetGyro();
    Drivetrain.getInstance().setSetpoint(Drivetrain.getInstance().getPosition());
    Drivetrain.getInstance().enable();
  }

  @Override
  public void autonomousPeriodic() {
    Scheduler.getInstance().run();
  }

  @Override
  public void teleopInit(){
    isAuto = false;
    Drivetrain.getInstance().enable();
  }

  @Override
  public void teleopPeriodic() {
    Scheduler.getInstance().run();
  }

  @Override
  public void testPeriodic() {
  }

  @Override
  public void disabledInit(){
    Drivetrain.getInstance().disable();
  }
}

File: src/main/java/org/minutebots/robot/Robot.kt
package org.minutebots.robot

import edu.wpi.first.wpilibj.TimedRobot
import edu.wpi.first.wpilibj.command.Scheduler
import org.minutebots.robot.hardwareconfigurations.*
import org.minutebots.robot.subsystems.Drivetrain
import org.minutebots.robot.utilities.VisionCommunication

class Robot : TimedRobot() {


    override fun robotInit() {
        hardwareManager.init()
        hardwareManager.closeSolenoids()
    }

    override fun robotPeriodic() {
        //VisionCommunication.getInstance().update();
    }

    override fun disabledPeriodic() {
        println(VisionCommunication.instance.angle)
    }

    override fun autonomousInit() {
        isAuto = true
        Drivetrain.instance.resetGyro()
        VisionCommunication.instance.highExposure()
        Drivetrain.instance.setSetpoint(Drivetrain.instance.getPosition())
        Drivetrain.instance.enable()
    }

    override fun autonomousPeriodic() {
        Scheduler.getInstance().run()
    }

    override fun teleopInit() {
        isAuto = false
        Drivetrain.instance.enable()
    }

    override fun teleopPeriodic() {
        Scheduler.getInstance().run()
    }

    override fun testPeriodic() {}

    override fun disabledInit() {
        Drivetrain.instance.disable()
    }

    companion object {

        var hardwareManager: HardwareManger = Asimov()
        var isAuto = false
    }
}
